src/table/src/metadata.rs

```diff
@@ -458,6 +459,118 @@ impl TableMeta {
         Ok(meta_builder)
     }
 
+    fn modify_columns(
```

Comment:
non-blocking: Add more tests to cover branches of the `modify_columns`(Maybe in another PR)